homeassistant/components/lamarzocco/services.py

```diff
@@ -0,0 +1,163 @@
+    {
+        vol.Required(CONF_DAY_OF_WEEK): vol.In(DAYS),
+        vol.Required(CONF_ENABLE): cv.boolean,
+    }
+)
+
+SET_AUTO_ON_OFF_TIMES_SCHEMA = CONFIG_ENTRY_SCHEMA.extend(
+    {
+        vol.Required(CONF_DAY_OF_WEEK): vol.In(DAYS),
+        vol.Required(CONF_HOUR_ON): vol.All(vol.Coerce(int), vol.Range(min=0, max=23)),
+        vol.Optional(CONF_MINUTE_ON, default=0): vol.All(
+            vol.Coerce(int), vol.Range(min=0, max=59)
+        ),
+        vol.Required(CONF_HOUR_OFF): vol.All(vol.Coerce(int), vol.Range(min=0, max=23)),
+        vol.Optional(CONF_MINUTE_OFF, default=0): vol.All(
+            vol.Coerce(int), vol.Range(min=0, max=59)
+        ),
+    }
+)
+
+
+async def __call_service(
+    func: Callable[..., Coroutine[Any, Any, Any]], *args: Any, **kwargs: Any
+) -> None:
+    """Call a service and handle exceptions."""
+    try:
+        await func(*args, **kwargs)
+    except (AuthFail, RequestNotSuccessful, TimeoutError) as ex:
+        raise HomeAssistantError("Service call encountered error: %s" % str(ex)) from ex
+
+
+def __get_coordinator(
+    hass: HomeAssistant, call: ServiceCall
+) -> LaMarzoccoUpdateCoordinator:
+    """Get the coordinator from the entry."""
+    entry_id: str = call.data[CONF_CONFIG_ENTRY]
+    entry: ConfigEntry | None = hass.config_entries.async_get_entry(entry_id)
+
+    if not entry:
+        raise ServiceValidationError(
+            f"Invalid config entry: {entry_id}",
+            translation_domain=DOMAIN,
+            translation_key="invalid_config_entry",
+            translation_placeholders={
+                "config_entry": entry_id,
+            },
+        )
+    if entry.state != ConfigEntryState.LOADED:
+        raise ServiceValidationError(
+            f"Config entry {entry.title} is not loaded",
+            translation_domain=DOMAIN,
+            translation_key="unloaded_config_entry",
+            translation_placeholders={
+                "config_entry": entry.title,
+            },
+        )
+    coordinator: LaMarzoccoUpdateCoordinator = hass.data[DOMAIN][entry_id]
+    return coordinator
+
+
+@callback
+def async_setup_services(hass: HomeAssistant) -> None:
+    """Create and register services for the La Marzocco integration."""
+
+    async def _set_auto_on_off_enable(service: ServiceCall) -> None:
+        """Service call to enable auto on/off."""
+
+        day_of_week = service.data[CONF_DAY_OF_WEEK]
+        enable = service.data[CONF_ENABLE]
+
+        coordinator = __get_coordinator(hass, service)
+
+        _LOGGER.debug("Setting auto on/off for %s to %s", day_of_week, enable)
+        await __call_service(
+            coordinator.lm.set_auto_on_off_enable,
+            day_of_week=day_of_week,
+            enable=enable,
+        )
+
+    async def _set_auto_on_off_times(service: ServiceCall) -> None:
+        """Service call to configure auto on/off hours for a day."""
+        day_of_week = service.data[CONF_DAY_OF_WEEK]
```

Comment:
Does this service not the same as the calendar entity in https://github.com/home-assistant/core/pull/108237?

Reply:
calendar platform is for display only, this is for setting only. As the app sets one schedule for mon-sun which applies to every week, editing one calendar entry, which then would edit all entries would be weird.